chemprop/cli/train.py

```diff
@@ -474,65 +477,86 @@ def validate_train_args(args):
 
 
 def normalize_inputs(train_dset, val_dset, args):
-    X_d_scaler, V_f_scaler, E_f_scaler, V_d_scaler = None, None, None, None
+    multicomponent = isinstance(train_dset, MulticomponentDataset)
+    num_components = train_dset.n_components if multicomponent else 1
 
-    if isinstance(train_dset, MulticomponentDataset):
-        d_xd = train_dset.datasets[0].d_xd
-        d_vf = sum(dset.d_vf for dset in train_dset.datasets)
-        d_ef = sum(dset.d_ef for dset in train_dset.datasets)
-        d_vd = sum(dset.d_vd for dset in train_dset.datasets)
-    else:
-        d_xd = train_dset.d_xd
-        d_vf = train_dset.d_vf
-        d_ef = train_dset.d_ef
-        d_vd = train_dset.d_vd
+    X_d_transform = None
+    V_f_transforms = [nn.Identity()] * num_components
+    E_f_transforms = [nn.Identity()] * num_components
+    V_d_transforms = [None] * num_components
+    graph_transforms = []
+
+    d_xd = train_dset.d_xd
+    d_vf = train_dset.d_vf
+    d_ef = train_dset.d_ef
+    d_vd = train_dset.d_vd
 
     if d_xd > 0 and not args.no_descriptor_scaling:
-        X_d_scaler = train_dset.normalize_inputs("X_d")
-        val_dset.normalize_inputs("X_d", X_d_scaler)
+        scaler = train_dset.normalize_inputs("X_d")
+        val_dset.normalize_inputs("X_d", scaler)
 
-        X_d_scaler = [X_d_scaler] if not isinstance(X_d_scaler, list) else X_d_scaler
+        scaler = scaler if not isinstance(scaler, list) else scaler[0]
 
-        scaler = X_d_scaler[0]
-        logger.info(f"Descriptors: loc = {scaler.mean_}, scale = {scaler.scale_}")
+        if scaler is not None:
+            logger.info(f"Descriptors: loc = {scaler.mean_}, scale = {scaler.scale_}")
+            X_d_transform = ScaleTransform.from_standard_scaler(scaler)
 
     if d_vf > 0 and not args.no_atom_feature_scaling:
-        V_f_scaler = train_dset.normalize_inputs("V_f")
-        val_dset.normalize_inputs("V_f", V_f_scaler)
+        scaler = train_dset.normalize_inputs("V_f")
+        val_dset.normalize_inputs("V_f", scaler)
 
-        V_f_scaler = [V_f_scaler] if not isinstance(V_f_scaler, list) else V_f_scaler
+        scalers = [scaler] if not isinstance(scaler, list) else scaler
 
-        for i, scaler in enumerate(V_f_scaler):
-            if scaler is not None:
-                logger.info(
-                    f"Atom features for mol {i}: loc = {scaler.mean_}, scale = {scaler.scale_}"
-                )
+        for i, scaler in enumerate(scalers):
+            if scaler is None:
+                continue
+
+            logger.info(f"Atom features for mol {i}: loc = {scaler.mean_}, scale = {scaler.scale_}")
```

Comment:
same goes for other logging spots as well- keep the logs as concise as possible.
```suggestion
            logger.info(f"Atom features for mol {i}: loc = {scaler.mean_:0.3f}, scale = {scaler.scale_:0.3f}")
```